mecanum_drive_controller/include/mecanum_drive_controller/mecanum_drive_controller.hpp

```diff
@@ -0,0 +1,137 @@
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  MECANUM_DRIVE_CONTROLLER__VISIBILITY_PUBLIC
+  controller_interface::CallbackReturn on_activate(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  MECANUM_DRIVE_CONTROLLER__VISIBILITY_PUBLIC
+  controller_interface::CallbackReturn on_deactivate(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  MECANUM_DRIVE_CONTROLLER__VISIBILITY_PUBLIC
+  controller_interface::return_type update_reference_from_subscribers(
+    const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+  MECANUM_DRIVE_CONTROLLER__VISIBILITY_PUBLIC
+  controller_interface::return_type update_and_write_commands(
+    const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+  using ControllerReferenceMsg = geometry_msgs::msg::TwistStamped;
+  using OdomStateMsg = nav_msgs::msg::Odometry;
+  using TfStateMsg = tf2_msgs::msg::TFMessage;
+  using ControllerStateMsg = control_msgs::msg::MecanumDriveControllerState;
+
+protected:
+  std::shared_ptr<mecanum_drive_controller::ParamListener> param_listener_;
+  mecanum_drive_controller::Params params_;
+
+  // used for chained controller
+  std::vector<std::string> state_joint_names_;
+
+  //Names of the references, ex: high level vel commands from MoveIt, Nav2, etc. used for preceding controller
+  std::vector<std::string> reference_names_;
+
+  // Command subscribers and Controller State, odom state, tf state publishers
+  rclcpp::Subscription<ControllerReferenceMsg>::SharedPtr ref_subscriber_ = nullptr;
+  realtime_tools::RealtimeBuffer<std::shared_ptr<ControllerReferenceMsg>> input_ref_;
+  rclcpp::Duration ref_timeout_ = rclcpp::Duration::from_seconds(0.0);
+
+  using OdomStatePublisher = realtime_tools::RealtimePublisher<OdomStateMsg>;
+  rclcpp::Publisher<OdomStateMsg>::SharedPtr odom_s_publisher_;
+  std::unique_ptr<OdomStatePublisher> rt_odom_state_publisher_;
+
+  using TfStatePublisher = realtime_tools::RealtimePublisher<TfStateMsg>;
+  rclcpp::Publisher<TfStateMsg>::SharedPtr tf_odom_s_publisher_;
+  std::unique_ptr<TfStatePublisher> rt_tf_odom_state_publisher_;
+
+  using ControllerStatePublisher = realtime_tools::RealtimePublisher<ControllerStateMsg>;
+  rclcpp::Publisher<ControllerStateMsg>::SharedPtr controller_s_publisher_;
+  std::unique_ptr<ControllerStatePublisher> controller_state_publisher_;
+
+  // override methods from ChainableControllerInterface
+  std::vector<hardware_interface::CommandInterface> on_export_reference_interfaces() override;
+
+  bool on_set_chained_mode(bool chained_mode) override;
+
+  Odometry odometry_;
+
+private:
+  // callback for topic interface
+  MECANUM_DRIVE_CONTROLLER__VISIBILITY_LOCAL
+  void reference_callback(const std::shared_ptr<ControllerReferenceMsg> msg);
+
+  double velocity_in_center_frame_linear_x;   // [m/s]
+  double velocity_in_center_frame_linear_y;   // [m/s]
+  double velocity_in_center_frame_angular_z;  // [rad/s]
```

Comment:
Global variables end with `_`

```suggestion
  double velocity_in_center_frame_linear_x_;   // [m/s]
  double velocity_in_center_frame_linear_y_;   // [m/s]
  double velocity_in_center_frame_angular_z_;  // [rad/s]
```